sentry_sdk/integrations/strawberry.py

```diff
@@ -0,0 +1,401 @@
+
+
+def _patch_views():
+    # type: () -> None
+    old_handle_errors_async = async_base_view.AsyncBaseHTTPView._handle_errors
+    old_handle_errors_sync = sync_base_view.SyncBaseHTTPView._handle_errors
+
+    def _sentry_patched_handle_errors_async(self, errors, response_data):
+        # type: (Any, List[GraphQLError], GraphQLHTTPResponse) -> None
+        old_handle_errors_async(self, errors, response_data)
+        _sentry_patched_handle_errors(self, errors, response_data)
+
+    def _sentry_patched_handle_errors_sync(self, errors, response_data):
+        # type: (Any, List[GraphQLError], GraphQLHTTPResponse) -> None
+        old_handle_errors_sync(self, errors, response_data)
+        _sentry_patched_handle_errors(self, errors, response_data)
+
+    def _sentry_patched_handle_errors(self, errors, response_data):
+        # type: (Any, List[GraphQLError], GraphQLHTTPResponse) -> None
+        hub = Hub.current
+        integration = hub.get_integration(StrawberryIntegration)
+        if integration is None:
+            return
+
+        if not errors:
+            return
+
+        with hub.configure_scope() as scope:
+            event_processor = _make_response_event_processor(response_data)
+            scope.add_event_processor(event_processor)
+
+        with capture_internal_exceptions():
+            for error in errors:
+                event, hint = event_from_exception(
+                    error,
+                    client_options=hub.client.options if hub.client else None,
+                    mechanism={
+                        "type": integration.identifier,
+                        "handled": False,
+                    },
+                )
+                hub.capture_event(event, hint=hint)
+
+    async_base_view.AsyncBaseHTTPView._handle_errors = (
+        _sentry_patched_handle_errors_async
+    )
+    sync_base_view.SyncBaseHTTPView._handle_errors = _sentry_patched_handle_errors_sync
+
+
+def _make_request_event_processor(execution_context):
+    # type: (ExecutionContext) -> EventProcessor
+
+    def inner(event, hint):
+        # type: (Dict[str, Any], Dict[str, Any]) -> Dict[str, Any]
+        with capture_internal_exceptions():
+            if _should_send_default_pii():
+                request_data = event.setdefault("request", {})
+                request_data["api_target"] = "graphql"
+
+                if not request_data.get("data"):
+                    request_data["data"] = {"query": execution_context.query}
+
+                    if execution_context.variables:
+                        request_data["data"]["variables"] = execution_context.variables
+                    if execution_context.operation_name:
+                        request_data["data"][
+                            "operationName"
+                        ] = execution_context.operation_name
+
+            elif event.get("request", {}).get("data"):
+                del event["request"]["data"]
```

Comment:
Made one slight change to it though: https://github.com/getsentry/sentry-python/pull/2393/commits/783152002244268784cab9b730722f1f0aac1e4e So that, if for some reason `event["request"]` is there and is `None`, we don't get an `AttributeError`.

Reply:
I have thought about this a bit more, and I think I would really prefer to switch to the try/except syntax, since I think it is much more readable for people looking at this code for the first time.

With the try/except block, the code reads in plain English as "If we want to send PII, send it, otherwise ensure we don't send any request data." It is easy to see what the code is doing.

With the `elif` statement, the code reads in plain English as "If we want to send PII, send it, otherwise check whether the event has a request (treat the request as an empty dictionary if it doesn't) and check whether the event's request has any data, and if it does, delete the event request data." In this situation, the code's objective is less obvious; only after thoroughly reading the contents of the elif condition and the statement it contains is it clear that the code here is designed to ensure the request has no data when we send the event.

Although the `elif` version is more concise, this concision comes at the expense of readability and understandability, so I think the try/except version would be preferable here.

Reply:
Ok, changed: https://github.com/getsentry/sentry-python/pull/2393/commits/2ccd1eb3125decec6e58689b7c203632e44a59ff
(The `TypeError` is there for the `event["request"] == None` case -- it'd throw that instead of the `AttributeError` I mentioned above.)